feat: загрузка датасета

sokuroff opened the pull request:
1. В FileHandler.py был добавлен метод для загрузки датасета. По умолчанию он сохраняет данные в data/raw
2. Добавлено игнорирование .csv файлов в .gitignore, чтобы гит не пытался залить их в github

sokuroff committed:
feat: download dataset

1. В FileHandler.py был добавлен метод для загрузки датасета. По умолчанию он сохраняет данные в data/raw
2. Добавлено игнорирование .csv файлов в .gitignore, чтобы гит не пытался залить их в github

Darkvoyant commented on src/FileHandler.py:
```diff
@@ -21,4 +24,80 @@ def read_file(path_to_file: str) -> pd.DataFrame:
-    
+
+    @staticmethod
+    def download_data(url: str = 'https://drive.fondsmena.ru/s/3fiHapYG8cpbafk/download',
```
Дефолтные параметры тут лучше не ставить, ссылку потом желательно вынести в какой-нибудь конфигурационный файл.

sokuroff replied:
поменял

Darkvoyant commented on src/FileHandler.py:
```diff
@@ -21,4 +24,80 @@ def read_file(path_to_file: str) -> pd.DataFrame:
+
+    @staticmethod
+    def download_data(url: str = 'https://drive.fondsmena.ru/s/3fiHapYG8cpbafk/download',
+                      save_dir: str = './data/raw/') -> None:
```
Тут бы тоже дефолтные параметры не ставила

Darkvoyant commented on src/FileHandler.py:
```diff
@@ -21,4 +24,80 @@ def read_file(path_to_file: str) -> pd.DataFrame:
+                    for chunk in r.iter_content(chunk_size=8192):
+                        f.write(chunk)
+
+            # распаковка
```
Я бы распаковку для архива сделала отдельным приватным методом

Darkvoyant commented on src/FileHandler.py:
```diff
@@ -21,4 +24,80 @@ def read_file(path_to_file: str) -> pd.DataFrame:
+            nested_archives = [p for p in save_dir_path.rglob('*.zip') if p != first_archive_path]
+
+            # если вложенных архивов нет, значит что-то пошло не так
+            if not nested_archives:
```
Я бы не выкидывала ошибку, если нет архива, так как он в целом может быть один. Мб лучше стоит проверять наличие csv файлов внутри?

Darkvoyant reviewed (changes requested):
Распаковку архивов я бы все же вынесла в отдельную функцию, которая возвращает путь до распакованной папки и сразу чистит за собой. В целом можно сделать цикл, в котором итерационно проходиться по всем zip файлам, пока они не кончатся. 

В общем я бы как-то поделила функцию, так как выглядит слишком нагружено

sokuroff committed:
fix: download dataset

1. Теперь функция работает так, что если после одной разархивации не обнаружено .csv файлов, то выдаётся ошибка;
2. убрано дефолтное значение url, target_dir;
3. метод был переименован в download_and_unpack_data чтобы отражать суть происходящего

Darkvoyant reviewed (approved):
Ок, пока пойдет
